tests/page/page-mouse.spec.ts

```diff
@@ -78,6 +78,38 @@ it('should dblclick the div', async ({ page, server }) => {
   expect(event.button).toBe(0);
 });
 
+it('should pointerdown the div with a custom button', async ({ page, server, browserName }) => {
+  await page.setContent(`<div style='width: 100px; height: 100px;'>Click me</div>`);
+  await page.evaluate(() => {
+    window['pointerdownPromise'] = new Promise(resolve => {
+      document.querySelector('div').addEventListener('pointerdown', event => {
+        resolve({
+          type: event.type,
+          detail: event.detail,
+          clientX: event.clientX,
+          clientY: event.clientY,
+          isTrusted: event.isTrusted,
+          button: event.button,
+          buttons: event.buttons,
+          pointerId: event.pointerId,
+        });
+      });
+    });
+  });
+  await page.mouse.click(50, 60, {
+    button: 'middle'
+  });
+  const event = await page.evaluate(() => window['pointerdownPromise']);
+  expect(event.type).toBe('pointerdown');
+  expect(event.detail).toBe(browserName === 'webkit' ? 1 : 0);
+  expect(event.clientX).toBe(50);
+  expect(event.clientY).toBe(60);
+  expect(event.isTrusted).toBe(true);
+  expect(event.button).toBe(1);
+  expect(event.buttons).toBe(4);
```

Comment:
It seems that we get 2 in webkit here. I think this is because the buttons mask map middle button to 2 here:
https://github.com/microsoft/playwright/blob/ea9475d37d885abca7e604ba42642a2975f50f7f/packages/playwright-core/src/server/webkit/wkInput.ts#L43-L44

This is different from the mapping used for firefox:
https://github.com/microsoft/playwright/blob/ea9475d37d885abca7e604ba42642a2975f50f7f/packages/playwright-core/src/server/firefox/ffInput.ts#L52-L53
and from the mapping that I've implemented in my other PR here:
https://github.com/microsoft/playwright/pull/10698/files#diff-0d5748af7b9e00da36d86779f7daf1d0fb86b749910be5ab759f4ebaad232538R112-R113

It also doesn't match the expected mapping described in the MDN:
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons#value

I've checked in Safari and it seems the right click maps correctly, as expected, to 2:
https://jsfiddle.net/8Lqyvs4m/1/

I can't currently verify the middle button cause I don't have access to the physical mouse right now but if right click maps to 2 then it would make sense that the middle button would map to 4.

So... I think this is a bug in the existing webkit mapping. Should I fix it as part of this PR?

Reply:
Let's see what happens on the other platforms. The numbers that we send to different browsers are different, nothing wrong with it - they all have their own enums. But you might be right with this one.

Reply:
> So... I think this is a bug in the existing webkit mapping. Should I fix it as part of this PR?

Yup, looks consistent on all ports, so safe to fix as a part of this PR!